kayttooikeus-service/src/main/java/fi/vm/sade/kayttooikeus/service/impl/UnohtunutSalasanaServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package fi.vm.sade.kayttooikeus.service.impl;
+
+import fi.vm.sade.kayttooikeus.dto.YhteystietojenTyypit;
+import fi.vm.sade.kayttooikeus.model.Henkilo;
+import fi.vm.sade.kayttooikeus.model.VarmennusPoletti;
+import fi.vm.sade.kayttooikeus.repositories.HenkiloDataRepository;
+import fi.vm.sade.kayttooikeus.repositories.VarmennusPolettiRepository;
+import fi.vm.sade.kayttooikeus.service.EmailService;
+import fi.vm.sade.kayttooikeus.service.TimeService;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import fi.vm.sade.kayttooikeus.service.UnohtunutSalasanaService;
+import fi.vm.sade.kayttooikeus.service.exception.DataInconsistencyException;
+import fi.vm.sade.kayttooikeus.service.external.OppijanumerorekisteriClient;
+import static fi.vm.sade.kayttooikeus.util.FunctionalUtils.ifPresentOrElse;
+import fi.vm.sade.kayttooikeus.util.YhteystietoUtil;
+import fi.vm.sade.oppijanumerorekisteri.dto.HenkiloDto;
+import fi.vm.sade.oppijanumerorekisteri.dto.YhteystietoTyyppi;
+import java.util.Set;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class UnohtunutSalasanaServiceImpl implements UnohtunutSalasanaService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UnohtunutSalasanaServiceImpl.class);
```

Comment:
Vain vinkkinä kun näin Villen käyttävän sitä jossain aiemmin. Jos ei tunnu hyödylliseltä niin ei oteta käyttöön.